raiden_contracts/utils/signature.py

```diff
@@ -8,15 +8,23 @@
 
 
 def sign(privkey: str, msg: bytes, v: int = 0) -> bytes:
-    assert isinstance(msg, bytes)
-    assert isinstance(privkey, str)
+    if not isinstance(msg, bytes):
+        raise TypeError("sign(): msg is not an instance of bytes")
+    if len(msg) != 32:
+        raise ValueError("sign(): msg has to be exactly 32 bytes")
+    if not isinstance(privkey, str):
+        raise TypeError("sign(): privkey is not an instance of str")
+    if v not in {0, 27}:
+        raise ValueError(f"sign(): got v = {v} expected 0 or 27.")
 
     pk = PrivateKey.from_hex(remove_0x_prefix(privkey))
-    assert len(msg) == 32
-
     sig: bytes = pk.sign_recoverable(msg, hasher=None)
     assert len(sig) == 65
 
+    pub = pk.public_key
+    recovered = PublicKey.from_signature_and_message(sig, msg, hasher=None)
+    assert pub == recovered
```

Comment:
I know you have a test for this already, but an additional `assert signature_test_contract.functions.verify(msg, sig).call() == pub_to_address` (maybe you can make this pretty) would give me some serious peace of mind here.

Reply:
I'll add it in this PR.

Reply:
Or not.  The problem is that we don't have access to any smart contract here.

Reply:
https://github.com/raiden-network/raiden-contracts/issues/1067 will keep track of this.